### PR TITLE
Fix Thaumcraft leaves lag

### PR DESCRIPTION
**Pull request by HoleFish**
Fixes https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/16092. Should be tested with https://github.com/GTNewHorizons/Random-Things/pull/6

### CODE REVIEW

File: src/main/java/com/mitchej123/hodgepodge/mixins/late/thaumcraft/MixinBlockMagicalLeaves.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    private MixinBlockMagicalLeaves(Material material) {
+        super(material);
+    }
+
+    /**
+     * @author HoleFish
+     * @reason Fix frequent ticking caused by old method
+     */
+    @Overwrite
+    public void updateTick(World worldIn, int x, int y, int z, Random random) {
```
Comment (Lyfts) [inline src/main/java/com/mitchej123/hodgepodge/mixins/late/thaumcraft/MixinBlockMagicalLeaves.java:33]: Does this method essentially just replace the worldIn.setBlock(...) in thaum with worldIn.setBlockMetadataWithNotify(...) or am I missing something?

Comment (HoleFish) [reply src/main/java/com/mitchej123/hodgepodge/mixins/late/thaumcraft/MixinBlockMagicalLeaves.java:33]: Yes. Ive tried redirection but it surprisingly didnt work. Maybe I should try harder;)

**Comment by Glease** [comment]
If it is so hard to get right with mixin I'd try asm. See https://github.com/Glease/TC4Tweaks/commit/c700d15b635b6e92fd171d7fe8fb0ffcacd0af2d

File: src/main/java/com/mitchej123/hodgepodge/mixins/late/thaumcraft/MixinBlockMagicalLeaves.java (hunk 1 above)
Comment (Lyfts) [reply src/main/java/com/mitchej123/hodgepodge/mixins/late/thaumcraft/MixinBlockMagicalLeaves.java:33]: Something like this should do the trick
```
@Redirect(
            method = "updateTick",
            at = @At(
                    value = "INVOKE",
                    target = "Lnet/minecraft/world/World;setBlock(IIILnet/minecraft/block/Block;II)Z",
                    remap = false))
    public boolean hodgepode$xxx(World world, int x, int y, int z, Block block, int meta, int flag) {}
```

Comment (HoleFish) [reply src/main/java/com/mitchej123/hodgepodge/mixins/late/thaumcraft/MixinBlockMagicalLeaves.java]: I tried exactly what you said 2 days ago and it did't work. Now I try again and it works well. Maybe my codes need a nap :trollface:

File: src/main/java/com/mitchej123/hodgepodge/mixins/late/thaumcraft/MixinBlockMagicalLog.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+public abstract class MixinBlockMagicalLog extends BlockRotatedPillar {
+
+    protected MixinBlockMagicalLog(Material p_i45425_1_) {
+        super(p_i45425_1_);
+    }
+
+    /**
+     * Old breakBlock() does not override the correct method.
+     */
+    @Override
```
Comment (eigenraven) [inline src/main/java/com/mitchej123/hodgepodge/mixins/late/thaumcraft/MixinBlockMagicalLog.java:22]: Did you mean `@Overwrite` here?

Comment (HoleFish) [reply src/main/java/com/mitchej123/hodgepodge/mixins/late/thaumcraft/MixinBlockMagicalLog.java:22]: No. `BlockMagicalLog` does not actually override `breakBlock()` and I override it here

**Review by eigenraven — APPROVED** [review]

**Pull request merged**